released/usecases/services_apps/sap_build_code/step1/main.tf

```diff
@@ -0,0 +1,328 @@
+# Setup destination
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "destination" {
+  subaccount_id = btp_subaccount.build_code.id
+  service_name  = "destination"
+  plan_name     = "lite"
+}
+# Get serviceplan_id for cicd-service with plan_name "default"
+data "btp_subaccount_service_plan" "destination" {
+  subaccount_id = btp_subaccount.build_code.id
+  offering_name = "destination"
+  name          = "lite"
+  depends_on    = [btp_subaccount_entitlement.destination]
+}
+# Create service instance
+resource "btp_subaccount_service_instance" "destination" {
+  subaccount_id  = btp_subaccount.build_code.id
+  serviceplan_id = data.btp_subaccount_service_plan.destination.id
+  name           = "destination"
+  depends_on     = [btp_subaccount_service_binding.cicd_service, data.btp_subaccount_service_plan.destination]
+  parameters = jsonencode({
+    HTML5Runtime_enabled = true
+    init_data = {
+      subaccount = {
+        existing_destinations_policy = "update"
+        destinations = [
+          # This is the destination to the cicd-service binding
+          {
+            Description                = "[Do not delete] SAP Continuous Integration and Delivery"
+            Type                       = "HTTP"
+            clientId                   = "${jsondecode(btp_subaccount_service_binding.cicd_service.credentials)["uaa"]["clientid"]}"
+            clientSecret               = "${jsondecode(btp_subaccount_service_binding.cicd_service.credentials)["uaa"]["clientsecret"]}"
+            "HTML5.DynamicDestination" = true
+            Authentication             = "OAuth2JWTBearer"
+            Name                       = "cicd-backend"
+            tokenServiceURL            = "${jsondecode(btp_subaccount_service_binding.cicd_service.credentials)["uaa"]["url"]}"
+            ProxyType                  = "Internet"
+            URL                        = "${jsondecode(btp_subaccount_service_binding.cicd_service.credentials)["url"]}"
+            tokenServiceURLType        = "Dedicated"
+          }
+        ]
+      }
+    }
+  })
+}
+
+# ------------------------------------------------------------------------------------------------------
+# APP SUBSCRIPTIONS
+# ------------------------------------------------------------------------------------------------------
+#
+# ------------------------------------------------------------------------------------------------------
+# Setup build-code
+# ------------------------------------------------------------------------------------------------------
+# Entitle
+resource "btp_subaccount_entitlement" "build_code" {
+  subaccount_id = btp_subaccount.build_code.id
+  service_name  = "build-code"
+  plan_name     = "standard"
+}
+# Subscribe
+resource "btp_subaccount_subscription" "build_code" {
```

Comment:
Generic question/remark: if using SAP Build Code, I can chose if what services/apps I want to leverage (at least accoring to the cost calculator).
This script creates the full range of apps and services that are part of SAP Build Code without giving choice to omit some services/apps in their specific setup. Maybe some boolean varaiables might make sense to allow this flexibility as provided by the booster

Reply:
Sounds great. Will make the necessary changes!

Reply:
Added two additional variables (`use_document_service` and `use_mobile_service`) to enable document service and mobile service. These two seem to be not mandatory for SAP Build Code.